pywinauto/recorder/win32/injector.py

```diff
@@ -0,0 +1,171 @@
+    def close_pipe(self):
+        if not self.h_pipe:
+            return
+        ctypes.windll.kernel32.FlushFileBuffers(self.h_pipe)
+        ctypes.windll.kernel32.DisconnectNamedPipe(self.h_pipe)
+        ctypes.windll.kernel32.CloseHandle(self.h_pipe)
+
+    def read_massage(self):
+        if not self.h_pipe:
+            return None
+        msg = ctypes.wintypes.MSG()
+        bytes_cnt = ctypes.c_ulong(0)
+        status = ctypes.windll.kernel32.ReadFile(self.h_pipe, msg, ctypes.sizeof(msg), ctypes.byref(bytes_cnt), None)
+        if status == 1 or bytes_cnt.value != 0:
+            return msg
+        return None
+
+    @property
+    def application(self):
+        """Return hooked application"""
+        return self.app
+
+    @staticmethod
+    def _get_process_handle(pid):
+        process_all_access =  win32defines.PROCESS_VM_OPERATION | win32defines.PROCESS_VM_READ | win32defines.PROCESS_VM_WRITE
+        return win32functions.OpenProcess(process_all_access, False, pid)
+
+    def _create_remote_thread(self, proc_address, arg_address, call_err_text = remote_call_error_str):
+        CreateRemoteThread = ctypes.windll.kernel32.CreateRemoteThread
+        c_function_adress = ctypes.WINFUNCTYPE(ctypes.c_ulong, ctypes.c_void_p)
+        thread_handle = CreateRemoteThread(self.h_process, None, 0, c_function_adress(proc_address), arg_address, 0, None)
+        if not thread_handle:
+            raise RuntimeError(call_err_text)
+        return thread_handle
+
+    def _virtual_alloc_for(self, buffer):
+        virtual_mem = win32defines.MEM_RESERVE | win32defines.MEM_COMMIT
+        address = ctypes.c_void_p(win32functions.VirtualAllocEx(self.h_process, 0, ctypes.sizeof(buffer), virtual_mem, win32defines.PAGE_READWRITE))
+        if not win32functions.WriteProcessMemory(self.h_process, address, ctypes.byref(buffer), ctypes.sizeof(buffer), 0):
+            raise AttributeError("Couldn't write data to process memory, check python acceess.")
+        return address
+
+    def _inject_dll_to_process(self):
+        # Convert dll path to unicode c_str
+        c_dll_path = ctypes.create_unicode_buffer(self.dll_path)
+        arg_address = self._virtual_alloc_for(c_dll_path)
+
+        # Get LoadLibraryW Address
+        GetModuleHandleW = ctypes.windll.kernel32.GetModuleHandleW
+        GetModuleHandleW.restype = ctypes.c_void_p
+        h_kernel32 = GetModuleHandleW(ctypes.c_wchar_p(unicode_string("kernel32.dll")))
+        h_loadlib = GetProcAddress(h_kernel32, byte_string("LoadLibraryW"))
+
+        # Now call CreateRemoteThread with entry point set to LoadLibraryW and pointer to DLL path as param
+        thread_handle = self._create_remote_thread(h_loadlib, arg_address, remote_call_injection_error_str)
+
+        ret = win32functions.WaitForSingleObject(thread_handle, remote_call_timeout)
+        if ret == win32defines.WAIT_TIMEOUT:
+            raise WaitError("Injection time out")
+
+    def _get_dll_proc_address(self, proc_name):
```

Comment:
Maybe `_get_dll_proc_address_w` is more informative or even `_get_dll_proc_address_unicode`.